### ISSUE TITLE
No index signature when a type has a symbol property

### ISSUE THREAD

**Issue by Conaclos** [Microsoft/TypeScript#22312]
**Search Terms:** "symbol", "no index signature", "symbol enumerable"

**Tested TypeScript versions:** `2.7.2` and `2.8.0-dev.20180302`

**Code**

```ts
const SYM = Symbol()
interface Type {
    prop: any,
    [SYM]: any
}
declare const x: Type
for (const p in x) {
    const v = x[p] // error, see below
}
```

**Expected behavior:**

No error when `noImplicitAny` is enabled.

**Actual behavior:**

The following error is reported:
```
error TS7017: Element implicitly has an 'any' type
because type 'Type' has no index signature.
```
Note that this error only occurs when noImplicitAny is turned on.

**Playground Link:** [play with it](https://www.typescriptlang.org/play/#src=const%20SYM%20%3D%20Symbol()%0D%0A%0D%0Ainterface%20Type%20%7B%0D%0A%20%20%20%20prop%3A%20any%2C%0D%0A%20%20%20%20%5BSYM%5D%3A%20any%0D%0A%7D%0D%0A%0D%0Adeclare%20const%20x%3A%20Type%0D%0A%0D%0Afor%20(const%20p%20in%20x)%20%7B%0D%0A%20%20%20%20const%20v%20%3D%20x%5Bp%5D%0D%0A%7D%0D%0A). Don't forget to enable **noImplicitAny**.

**Related Issues:**
The following issue could be related to:
https://github.com/Microsoft/TypeScript/issues/21912

**Temporary fix:**
For now you can use a type assertion:

```ts
const v = x[p as keyof Type]
```
Note that this works only if `Type` has at least one enumerable property.
